hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/HazelcastTypeCoercion.java

```diff
@@ -100,11 +98,6 @@ public boolean binaryComparisonCoercion(SqlCallBinding binding) {
         throw new UnsupportedOperationException("Should not be called");
     }
 
-    @Override
-    public boolean rowTypeCoercion(SqlValidatorScope scope, SqlNode query, int columnIndex, RelDataType targetType) {
```

Comment:
`rowTypeCoercion`, `querySourceCoercion`, and other similar methods are added here to ensure that we never rely on Calcite's coercion logic, but instead provide our own. Removal of these methods means that we again rely on Calcite, breaking the whole idea of controllable operator behavior. 

Instead, we should ensure that these functions are never invoked by providing the proper type checking implementations to the required operators.

Reply:
Do we have a final decision about `querySourceCoercion` and `rowTypeCoercion` methods? If we are going to use the Calcite implementation, can we leave the overridden methods, delegate from them to Calcite, but add a comment, explaining the motivation?